### PR TITLE
Release cleanup

### PR DESCRIPTION
**Pull request by archigup**
Updates version numbers, README.md documentation links, and adds C++ guards in core_json.h.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,7 +68,19 @@ gcc -I source/include example.c source/core_json.c -o example
...
+
+### Existing documentation
+For pre-generated documentation, please see the documentation linked in the locations below:
+
+| Location |
+| :-: |
+| [AWS IoT Device SDK for Embedded C](https://github.com/aws/aws-iot-device-sdk-embedded-C#releases-and-documentation) |
+| [FreeRTOS.org](https://freertos.org/Documentation/api-ref/coreJSON/docs/doxygen/output/html/index.html) |
+
+Note that the latest included version of the AWS IoT Device Defender library may differ across repositories.
```
Comment (muneebahmed10) [inline README.md:81]:
```suggestion
Note that the latest included version of the coreJSON library may differ across repositories.
```

**Review by muneebahmed10 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Change Log for coreJSON Library
 
+## v3.0.2 (July 2021)
+- [#100](https://github.com/FreeRTOS/coreJSON/pull/100) Fix overflow in skipOneHexEscape().
```
Comment (aggarw13) [inline CHANGELOG.md:4]: As there is a bugfix update, should this undergo a minor version update instead of a patch version change?

**Comment by aggarw13** [comment]
The `manifest.yml` file needs to be updated as well with the new version

**Review by aggarw13 — APPROVED** [review]

**Review by muneebahmed10 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (xortothemoon) [reply README.md]: copy pasta gangstas 4 lyf

**Review by n9wxu — APPROVED** [review]

**Pull request merged**